app/code/community/Sign2pay/Payment/Helper/Data.php

```diff
@@ -98,4 +157,50 @@ public function setStatusOnOrder($order, $status_code)
         $order->setState($status['state'], $status['status']);
     }
 
+    /**
+     * Prepare and return initial Sign2Pay request
+     * @todo device unical id
+     *
+     * @return string
+     */
+    public function getSign2PayInitialRequest(){
+        $quote = Mage::getSingleton('checkout/session')->getQuote();
+        
+        Mage::log($quote->getData());
+        $amount = $quote['grand_total']*100;
+        $amount = preg_replace('/[^0-9]/', '', $amount);
+        $ref_id = $this->sign2PayCheckoutHash($quote['reserved_order_id']);
+
+        $billing = $quote->getBillingAddress();
+
+        $baseUrl = 'https://app.sign2pay.com/oauth/authorize';
+        $client_id = $this->getSign2payClientId();
+        $redirect_uri = $this->getRedirectUri();
+
+        $scope = 'payment';
+        $state = $this->userStateHash();
+        $response_type = 'code';
+        $device_uid = 'test';
+
+        $query = http_build_query(array(
+            'client_id' => $client_id,
+            'redirect_uri' => $redirect_uri,
+            'scope' => $scope,
+            'amount' => $amount,
+            'state' => $state,
+            'ref_id' => $ref_id,
+            'response_type' => $response_type,
+            'device_uid' => $device_uid,            
+            'user_params[identifier]' => $billing['email'],
+            'user_params[first_name]' => $billing['firstname'],
+            'user_params[last_name]' => $billing['lastname'],
+            'user_params[address]' => $billing['street'],
```

Comment:
@magently can you include the house number as well? It will give more accurate risk insight.